src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -249,189 +315,272 @@ export const PersonalInfo: React.FC<IEditAuthorProps> = ({
+              )}
+            </Box>
+            <TextField
+              multiline
+              minRows={12}
+              variant="outlined"
+              required
+              fullWidth
+              name="bio"
+              value={newAuthorValues.bio}
+              onChange={inputChangeHandler}
+              onBlur={blurHandler}
+            />
+          </Grid>
         </Grid>
-      </Grid>
 
-      <Box className={classes.ButtonBox}>
-        {author && !isCurrentUser && (
-          <CancelButton
-            label={i18n.t(langTokens.common.cancelChanges)}
-            onClick={() => window.close()}
+        <Box className={classes.ButtonBox}>
+          {author && !isCurrentUser && (
+            <CancelButton
+              label={i18n.t(langTokens.common.cancelChanges)}
+              onClick={() => window.close()}
+            />
+          )}
+          <BasicButton
+            disabled={isSaveDisabled}
+            type="sign"
+            label={i18n.t(langTokens.common.acceptChanges)}
+            className={classes.BasicButton}
+            onClick={buttonClickHandler}
           />
-        )}
-        <BasicButton
-          disabled={isSaveDisabled}
-          type="sign"
-          label={i18n.t(langTokens.common.acceptChanges)}
-          className={classes.BasicButton}
-          onClick={buttonClickHandler}
-        />
-      </Box>
-    </form>
+        </Box>
+      </form>
+      {isAdmin && author && !isCurrentUser && (
+        <Container>
+          {toggleButton && (
+            <>
+              {!openForm && (
+                <>
+                  <Box className={classes.ButtonBox}>
+                    <Typography variant="h5">
+                      {i18n.t(langTokens.admin.activateOrCreateUserAccount)}
+                    </Typography>
+                  </Box>
+                  <Box className={classes.ButtonBox}>
+                    <UserAccountButton
+                      type="activate"
+                      label={i18n.t(langTokens.admin.activateExistingAccount)}
+                      onClick={handleChangeEnabled}
+                      disabled={isLoading}
+                    />
+                    <UserAccountButton
+                      type="create"
+                      disabled={isLoading}
+                      label={i18n.t(langTokens.admin.createUserAccount)}
+                      onClick={() => setOpenForm(true)}
+                    />
+                  </Box>
+                </>
+              )}
+              {openForm && (
+                <Box className={classes.ButtonBox}>
+                  <CreateUserAccountForm
+                    usersEmails={usersEmails}
+                    onClick={handleActivateUser}
+                    isLoading={isLoading}
+                  />
+                </Box>
+              )}
+            </>
+          )}
+          <Box className={classes.ButtonBox}>
+            {enabled && status === UserStatus.ACTIVE && (
```

Comment:
having multiple complex mutually exclusive conditions with ` ... && ...` is really tricky and risky - we may end up with multiple conditions evaluated to `true` at some moment. another concern is we don't have default value(when neither of our conditions matches) - which may be desired or not, so is confusing to the reader.

so instead of putting this into JSX, we could calculate `buttonState` variable so it will be guaranteed only single block to be rendered. It may be done with 
```
let buttonState;
if (..) { buttonState = '...' }  // only single case will be applied
else if ( ... ) { buttonState = ' ...' ` 
else if ... 
else {
  buttonState = 'hidden'; // example of default case
}
```
or it can be done with a separate function with early exit pattern in it; this will allow us to stop checking for specific variables at some point because we will be sure it has some specific value:

```
function getButtonState({enabled, status, toggleButton}) {
  if (enabled && status === UserStatus.ACTIVE) {
    return 'deactivate';
  }
  if (enabled && status !== UserStatus.ACTIVE) {
    return 'hidden';
  }
 // from here and down `enabled === false` and we don't need to check it any more
  .....  
  return 'hidden'; // default case
}
```

this "skipping condition check" is not a main goal, but depending on specific scenario can heavily simplify the conditions tree

Reply:
Sorry, i don`t understand how i can apply this, i can do separate if statments with return jsx or explain me more about this technique in our case, please

Reply:
I'm sorry for over detailed explanation, just want to give all the reasoning

So final goal is to make code which:

1. guarantees that no more than single piece of JSX will be rendered under any conditions
2. clearly outputs a default case when no condition matches - or explicitly shows that we don't need anything to be shown

Current version of 

```
{condition1 && condition2 && (JSX1)}
{condition3 && !condition2 && (JSX2)}
{/* etc */}
```

does not help with both. And I see next options here:

Option 1. Introducing single state-related variable right in the render logic:

```
let buttonTypeToShow: ActionButtonType = 'nothing';
if (condition1 && condition2) {
  buttonTypeToShow = 'deactivate';
} else if (condition1 && !condition2) {
  buttonTypeToShow = 'activate';
} else if (...) 
.....
  {buttonTypeToShow === 'activate' && (...)}
  {buttonTypeToShow === 'deactivate' && (....)}
```
It's clearly only single case satisfies and we can see that default value for `buttonTypeToShow` does not render anything - intentionally, not accidetnally.

Option 2. Introducing separate component with early exit pattern

```
function PersonalInfoActionButton({ condition1, condition2, condition3, onClick, children }) {
  if (condition1 && condition2) {
    return (JSX1);
  }
  if (condition1 && !condition2) {
    return (JSX2);
  }
  ....
 return null; // if nothing match we don't render anything
}
```

Option 3. The same as option 2 but without introducing separate component. IMO this is the worst way among those three - we cannot just read rendering part of `<PersonalInfo>` and have to navigate back and forth across the functions to understand what will be final result. But still, it saves us from case when mutliple conditions matches and few different buttons are rendered side-by-side:

```
function PersonalInfo({...}) {
  const renderButton = () => {
    if (condition1 && condition2) { // 
      return (JSX1);
    }
    if (condition1 && !condition2) {
      return (JSX2);
    }
    ....
   return null; // if nothing match we don't render anything
  }

 ....
  return (
  ....
    {renderButton()}
  ....
}
```